azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -606,18 +612,21 @@ private void runJob() {
     }
 
     // If the job is still running, set the status to Success.
-    if (!Status.isStatusFinished(node.getStatus())) {
-      changeStatus(Status.SUCCEEDED);
+    if (!Status.isStatusFinished(finalStatus)) {
+      finalStatus = changeStatus(Status.SUCCEEDED);
     }
+    return finalStatus;
   }
 
-  private void changeStatus(Status status) {
+  private Status changeStatus(Status status) {
     changeStatus(status, System.currentTimeMillis());
+    return status;
   }
 
-  private void changeStatus(Status status, long time) {
+  private Status changeStatus(Status status, long time) {
```

Comment:
It's used like this:

```java
    if (Status.isStatusFinished(nodeStatus)) {
      quickFinish = true;
    } else if (nodeStatus == Status.DISABLED) {
      nodeStatus = changeStatus(Status.SKIPPED, time);
      quickFinish = true;
    } else if (this.isKilled()) {
      nodeStatus = changeStatus(Status.KILLED, time);
      quickFinish = true;
    }
```

So it's just to do with less lines inside the if-else blocks. Alternative would look like:

```java
    if (Status.isStatusFinished(nodeStatus)) {
      quickFinish = true;
    } else if (nodeStatus == Status.DISABLED) {
      nodeStatus = Status.SKIPPED;
      changeStatus(nodeStatus, time);
      quickFinish = true;
    } else if (this.isKilled()) {
      nodeStatus = Status.KILLED;
      changeStatus(nodeStatus, time);
      quickFinish = true;
    }
```

I find the first one easier to read.